vanilla/src/api/AuthService.ts

```diff
@@ -0,0 +1,35 @@
+import { Auth, GoogleAuthProvider, signInWithPopup, signOut } from 'firebase/auth';
+
+/**
+ * Class for working with authorization.
+ */
+class AuthService {
+  /** Method to get isUserAuthorized flag. */
+  public get isUserAuthorizedGetter(): boolean {
+    return this.isUserAuthorized;
+  }
+
+  /** Flag indicating whether the user is logged in. */
+  private isUserAuthorized = false;
```

Comment:
You use `get` / `set` properties as common class fields:
```
isAuth = authService.isUserAuthorized; // get
authService.isUserAuthorized = true; // set
```
so in general we don't need to mark them either as *getter* or *setter*. For other application parts `isUserAuthorized` is one `AuthService` property.

```suggestion
private _isUserAuthorized = false;

public get isUserAuthorize(): boolen {
    this._isUserAuthorized;
}
```
I think this way it's a bit better